src/neo/SmartContract/Native/StdLib.cs

```diff
@@ -137,7 +143,63 @@ public static string Base58Encode(byte[] data)
         [ContractMethod(CpuFee = 1 << 12)]
         public static byte[] Base58Decode(string s)
         {
+            CheckInput(s);
             return Base58.Decode(s);
         }
+
+        [ContractMethod(CpuFee = 1 << 12)]
+        private static int MemoryCompare(byte[] str1, byte[] str2)
+        {
+            CheckInput(str1);
+            CheckInput(str2);
+            return Math.Sign(str1.AsSpan().SequenceCompareTo(str2));
+        }
+
+        [ContractMethod(CpuFee = 1 << 12)]
+        private static int MemorySearch(byte[] mem, byte[] value)
+        {
+            return MemorySearch(mem, value, 0, false);
+        }
+
+        [ContractMethod(CpuFee = 1 << 12)]
+        private static int MemorySearch(byte[] mem, byte[] value, int start)
+        {
+            return MemorySearch(mem, value, start, false);
+        }
+
+        [ContractMethod(CpuFee = 1 << 12)]
+        private static int MemorySearch(byte[] mem, byte[] value, int start, bool backward)
+        {
+            CheckInput(mem);
```

Comment:
Can I move this method to Attribute based?

Reply:
How?

Reply:
Take a look to https://github.com/neo-project/neo/pull/2424